src/components/App/DisplayPuzzleInstances/index.tsx

```diff
@@ -0,0 +1,137 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+import styles from './DisplayPuzzleInstances.module.scss';
+import { PuzzleCustom } from '../../../types/api/puzzles/puzzle';
+import { getAllPuzzleInstances } from '../../../services';
+import { PuzzleInstance } from '@prisma/client';
+import { Button } from '../../Global';
+import { QRGenerator } from '../../App';
+import toast from 'react-hot-toast';
+
+type DisplayPuzzleInstancesProps = {
+  puzzlesList: PuzzleCustom[];
+};
+
+const DisplayPuzzleInstances = ({
+  puzzlesList
+}: DisplayPuzzleInstancesProps) => {
+  const [puzzleId, setPuzzleId] = useState<string>('');
+  const [puzzleInstances, setPuzzleInstances] = useState<PuzzleInstance[]>([]);
+  const [displayPuzzleInstances, setDisplayPuzzleInstances] = useState<
+    PuzzleInstance[]
+  >([]);
+
+  const getPuzzleInstances = async () => {
+    const retrivedPuzzleInstances = await getAllPuzzleInstances();
+    setPuzzleInstances(retrivedPuzzleInstances);
+  };
+
+  useEffect(() => {
+    getPuzzleInstances();
+  }, []);
+
+  useEffect(() => {
+    getPuzzleInstances();
+  }, [puzzlesList]);
+
+  useEffect(() => {
+    if (!puzzleInstances || !setDisplayPuzzleInstances) return;
+
+    if (!puzzleId || puzzleId === '') {
+      setDisplayPuzzleInstances(puzzleInstances);
+    } else {
+      setDisplayPuzzleInstances(
+        puzzleInstances.filter(pzl => pzl.puzzleId.toString() === puzzleId)
+      );
+    }
+  }, [puzzleId, puzzleInstances, setDisplayPuzzleInstances]);
+
+  return (
+    <div className={styles.puzzleInstancesWrapper}>
+      {/* HEADER */}
+      <div className={styles.puzzleInstancesHeader}>
+        <div className={styles.puzzleInstancesHeaderTitle}>
+          <h3> See Locations </h3>
+        </div>
+        {/* Select puzzle */}
+        <div className={styles.puzzleInstancesHeaderSelectWrapper}>
+          {puzzlesList.length > 0 ? (
+            <select
+              className={styles.puzzleInstancesHeaderSelect}
+              value={puzzleId}
+              onChange={e => setPuzzleId(e.currentTarget.value)}
+            >
+              <option value="" key="all-puzzles">
+                All puzzles
+              </option>
+              {puzzlesList.map((puzzle: PuzzleCustom, index: number) => (
+                <option value={puzzle.id} key={`puzzle${index}`}>
+                  {puzzle.name}
+                </option>
+              ))}
+            </select>
+          ) : (
+            <select
+              className={styles.puzzleInstancesHeaderSelect}
+              value="No puzzle available"
+            ></select>
+          )}
+        </div>
+      </div>
+
+      {/* BODY */}
+      <div className={styles.puzzleInstancesBody}>
+        {/* 3 cols table title */}
```

Comment:
Why not use an actual `<table>` here though?

Reply:
To me, it's harder to have custom format for this

Reply:
I mean css